drivers/net/dsa/lantiq_gsw_mdio.c

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * MaxLinear / Lantiq / Intel GSW switch driver for external MDIO-managed parts.
+ * Currently only supports the GSW120 & GSW125.
+ * 
+ * See lantiq_gswip_core.c for additional information.
+ *
+ * Copyright (C) 2010 Lantiq Deutschland
+ * Copyright (C) 2012 John Crispin <john@phrozen.org>
+ * Copyright (C) 2017 - 2019 Hauke Mehrtens <hauke@hauke-m.de>
+ * Copyright (C) 2022 Reliable Controls Corporation, Harley Sims <hsims@reliablecontrols.com>
+ */
+
+/* TODO WARP-5829: determine how many of these includes I can delete */
+#include <linux/clk.h>
+#include <linux/delay.h>
+#include <linux/etherdevice.h>
+#include <linux/firmware.h>
+#include <linux/if_bridge.h>
+#include <linux/if_vlan.h>
+#include <linux/iopoll.h>
+#include <linux/mfd/syscon.h>
+#include <linux/module.h>
+#include <linux/of_mdio.h>
+#include <linux/of_net.h>
+#include <linux/of_platform.h>
+#include <linux/phy.h>
+#include <linux/phylink.h>
+#include <linux/platform_device.h>
+#include <linux/regmap.h>
+#include <linux/reset.h>
+#include <net/dsa.h>
+#include <dt-bindings/mips/lantiq_rcu_gphy.h>
+
+#include "lantiq_gsw.h"
+#include "lantiq_pce.h"
+
+static u32 gsw_mdio_read(struct gswip_priv *priv, void *address)
+{
+	/* TODO WARP-5829: write this function */
+	return 0;
+}
+
+static void gsw_mdio_write(struct gswip_priv *priv, u32 val, void *address)
+{
+	/* TODO WARP-5829: write this function */
+}
+
+static const struct gsw_ops gsw_mdio_ops = {
+	.read = gsw_mdio_read,
+	.write = gsw_mdio_write,
+};
+
+/*-------------------------------------------------------------------------*/
+
+static int gsw_mdio_probe(struct mdio_device *pmdiodev)
+{
+	/* TODO WARP-5829: write this function */
+	printk("!RCC: GSW MDIO probe func invoked.\n");
```

Comment:
I've confirmed I see these strings in the kernel message buffer on boot when my device tree has an MDIO-bus sub-node that specifies the new "maxlinear,gsw12x" compatible string.